docs/annotations.md

```diff
@@ -0,0 +1,202 @@
+```c++
+// TEMPLATE: LEGO1 0x10035790
+// ?_Construct@@YAXPAPAVROI@@ABQAV1@@Z
+```
+
+### Annotation types
+
+#### `FUNCTION`
+
+Functions with a reasonably complete implementation which are not templated or synthetic (see below) should be annotated with `FUNCTION`. It is preferable to annotate the function's implementation directly.
+
+#### `STUB`
+
+Functions with no or a very incomplete implementation should be annotated with `STUB`. These will not be compared to the original assembly.
+
+```c++
+// STUB: LEGO1 0x10011d50
+LegoCameraController::LegoCameraController()
+{
+  // TODO
+}
+```
+
+#### `TEMPLATE`
+
+Templated functions should be annotated with `TEMPLATE`. Since the goal is to eventually have a full accounting of all the functions present in the binaries, please make an effort to find and annotate every function of a templated class.
+
+```c++
+// TEMPLATE: LEGO1 0x100c0ee0
+// list<MxNextActionDataStart *,allocator<MxNextActionDataStart *> >::_Buynode
+
+// TEMPLATE: LEGO1 0x100c0fc0
+// MxStreamListMxDSSubscriber::~MxStreamListMxDSSubscriber
+
+// TEMPLATE: LEGO1 0x100c1010
+// MxStreamListMxDSAction::~MxStreamListMxDSAction
+```
+
+#### `SYNTHETIC`
+
+Synthetic functions should be annotated with `SYNTHETIC`. A synthetic function is generated by the compiler; most common is the "scalar deleting destructor" found in virtual tables. Other cases include default destructors and assignment operators. Note: `SYNTHETIC` takes precedence over `TEMPLATE`.
+
+```c++
+// SYNTHETIC: LEGO1 0x10003210
+// Helicopter::`scalar deleting destructor'
+
+// SYNTHETIC: LEGO1 0x100c4f50
+// MxCollection<MxRegionLeftRight *>::`scalar deleting destructor'
+
+// SYNTHETIC: LEGO1 0x100c4fc0
+// MxList<MxRegionLeftRight *>::`scalar deleting destructor'
+```
+
+#### `LIBRARY`
+
+Functions located in 3rd party libraries should be annotated with `LIBRARY`. This can be useful for working towards a full accounting of all the functions present in the binaries.
+
+```c++
+// LIBRARY: ISLE 0x4061b0
+// _MemPoolInit@4
+
+// LIBRARY: ISLE 0x406520
+// _MemPoolSetPageSize@8
+
+// LIBRARY: ISLE 0x406630
+// _MemPoolSetBlockSizeFS@8
+```
+
+
+## Virtual tables
+
+Classes with a virtual table should be annotated using the `VTABLE` marker, which includes the module name and address of the virtual table:
+```c++
+// VTABLE: LEGO1 0x100dc900
+class MxEventManager : public MxMediaManager {
+public:
+	MxEventManager();
+	virtual ~MxEventManager() override;
+
+	virtual void Destroy() override;                                     // vtable+0x18
+	virtual MxResult Create(MxU32 p_frequencyMS, MxBool p_createThread); // vtable+0x28
+    // ...
+}
+```
+While not relevant to the `reccmp` scripts, we made a rule in [isledecomp](https://github.com/isledecomp/isle) to annotate functions with a comment indicating their relative offset for convenience.
+
+
+
+## Class size
+
+**TODO: Is any of this relevant to reccmp or only to isledecomp?**
```

Comment:
This is specific to isledecomp.
However, this habit is useful for quick feedback so perhaps mention it in an extra section where you can list various decompilation hints?
Another hint could be to introduce `undefinedX` data types to make the difference between "known" data types (such as `uint8_t`, `char`, `uint8_t*`, `void*`, `uint32_t`, `int`) and unknown ones.